### PR TITLE
[SYCL] Fixed issues reported by a static verifier

### PR DESCRIPTION
**Pull request by bso-intel**

### CODE REVIEW

File: sycl/source/detail/graph_impl.cpp [hunk 1]
```diff
@@ -572,8 +577,10 @@ bool modifiable_command_graph::end_recording(
 
 executable_command_graph::executable_command_graph(
     const std::shared_ptr<detail::graph_impl> &Graph, const sycl::context &Ctx)
-    : MTag(rand()),
-      impl(std::make_shared<detail::exec_graph_impl>(Ctx, Graph)) {
+    : impl(std::make_shared<detail::exec_graph_impl>(Ctx, Graph)) {
+  std::default_random_engine randEngine;
+  std::uniform_int_distribution<int> dist(0, RAND_MAX);
+  MTag = dist(randEngine);
```
Comment (EwanC) [inline sycl/source/detail/graph_impl.cpp:583]: `MTag` is currently unused by the implementation, I think a better fix here might be to just remove the member variable and avoiding making the code more verbose with the explicit rand initalization.

**Review by EwanC — COMMENTED** [review]
Looking good in general, but have a suggestion for an alternative solution to one of the warnings.

File: sycl/source/detail/graph_impl.cpp (hunk 1 above)
Comment (maarquitos14) [reply sycl/source/detail/graph_impl.cpp:583]: +1, unless we plan to give `MTag` any use in the future.

**Review by maarquitos14 — APPROVED** [review]

**Review by EwanC — APPROVED** [review]

**Comment by bso-intel** [comment]
@intel/llvm-gatekeepers 
Please merge

File: sycl/source/detail/graph_impl.cpp [hunk 2]
```diff
@@ -507,27 +509,29 @@ modifiable_command_graph::finalize(const sycl::property_list &) const {
 
 bool modifiable_command_graph::begin_recording(queue &RecordingQueue) {
   auto QueueImpl = sycl::detail::getSyclObjImpl(RecordingQueue);
+  if (QueueImpl) {
```
Comment (aelovikov-intel) [inline sycl/source/detail/graph_impl.cpp:512]: Why this and not an `assert`?

Comment (bso-intel) [reply sycl/source/detail/graph_impl.cpp:521]: I am not sure if we should expect non-null QueueImpl from getSyclObjImpl().
Coverity mentioned that it is possibly a nullptr.

Comment (aelovikov-intel) [reply sycl/source/detail/graph_impl.cpp:521]: Well, if it can be a `nullptr`, then we need a test for that scenario as part of this PR.

Comment (bso-intel) [reply sycl/source/detail/graph_impl.cpp:521]: It is not ideal to create a test for the internal API getSyclObjImpl().
I can change this condition to assert(QueueImpl);
Would that satisfy your request?

Comment (aelovikov-intel) [reply sycl/source/detail/graph_impl.cpp:521]: Yes, I'd rather have assert/SIGSEGV then pretend that the code is supported only to fail much later in an obscure place.

Comment (bso-intel) [reply sycl/source/detail/graph_impl.cpp]: Done.

**Review by steffenlarsen — APPROVED** [review]

**Comment by steffenlarsen** [comment]
HIP issue reported in https://github.com/intel/llvm/issues/10460.

**Pull request merged**